config/serverless.security.yml

```diff
@@ -17,6 +17,12 @@ xpack.securitySolutionServerless.productTypes:
     { product_line: 'endpoint', product_tier: 'complete' },
   ]
 
+xpack.securitySolution.settings: {
+  sideNavEnabled: false,
+  ILMEnabled: false,
+  ESQLEnabled: false
```

Comment:
From what I know, ESQL is not available for serverless in a general way, so this PR unifies all those places you mentioned under the same setting.
If, at some point, ESQL is available for serverless, just turning this `true` will make the entire application enable ESQL features (they are already implemented).
What is the advantage of having separate settings?

Reply:
> If, at some point, ESQL is available for serverless, just turning this true will make the entire application enable ESQL features (they are already implemented).
> What is the advantage of having separate settings?

There is no guarantee, all features will be made available at the same time. All Security Soluton ES|QL features were developed independently, by different teams and it's GA availability on Serverless can vary depend on Product's needs, received feedback during testing, any possibly discovered issues, etc. Moreover, if we would like to test or make available only one feature for preview, granularity is very handy is such situations

Reply:
> now that we have settings in xpack.securitySolution.*. What to do with the new config values here? Should they be added into the same level as settings, or be added to .settings? i.e. xpack.securitySolution.myNewConfigValue vs xpack.securitySolution.settings.myNewConfigValue

Not sure to understand the question, we can add the new setting to `xpack.securitySolution.settings` config as described, or we can create a new config entry with a new schema if _settings_ do not adjust to your needs.

Reply:
> if we would like to test or make available only one feature for preview, granularity is very handy is such situations.

In my opinion, when ESQL is available for serverless, the limitation will no longer be there so we should enable this setting. And then, if some part of the app is still under development or not working properly, we can use experimentalFlags to disable it while we work on that.

In case we don't want a specific ESQL feature in some part of the Security app to be available on serverless, for some reason other than being a WIP, like a Product decision, or another kind of limitation. Then I agree, we could use _settings_, but I would create the setting flag at that moment. I don't think we should assume this scenario right now.

Reply:
the question was, what's the general purpose of settings object - should new values be added here on the root level of xpack. securitySolution.*? Since settings name can be seemed very general, so basically anything can be put there.

But after your proposal to change name to offeringsSettings - this question is basically answered

Reply:
Is there an issue in adding a granular flag for a feature?

 As I mentioned different teams works separately on each feature, and generally would like to have a control over it. If these settings doesn't allow granular access, so I think we need to come up with another way of doing this. We need a convenient way, with very little configuration to regulate access to a features depends on our needs: GA, demo preview, testing, etc. I don't see how having this granularity can negatively impact app. 

Also, I noted ES|QL timeline is available on serverless app, from this branch

<img width="2531" alt="Screenshot 2023-10-05 at 13 08 00" src="https://github.com/elastic/kibana/assets/92328789/33ad23a2-eb86-4237-a6d2-066553f44b9f">

@michaelolo24 , should it be hidden?

Reply:
> But after your proposal to change name to offeringsSettings - this question is basically answered
Yup, okay I'll rename it then 👍